### PR TITLE
Add auto detect resolution

### PR DESCRIPTION
**Pull request by 0lvin**
* Add auto detect resolution #355
* code cleanups for make cppcheck little more happy
* additional sanitize checks 
* move code patches authors from code comments to about menu

### CODE REVIEW

File: src/client/refresh/soft/sw_main.c [hunk 1]
```diff
@@ -1469,6 +1470,9 @@ static int RE_PrepareForWindow(void)
 	return flags;
 }
 
+// Declared in vid/header/ref.h
+refexport_t	re;
```
Comment (DanielGibson) [inline src/client/refresh/soft/sw_main.c]: This is not supposed to be a global variable in the ref DLLs.
It *is* a global variable in the executable (see vid.c) (and is set from the return value of GetRefAPI() from the ref DLL)

File: src/client/refresh/gl3/gl3_sdl.c [hunk 2]
```diff
@@ -341,3 +341,27 @@ void GL3_ShutdownContext()
...
+ * Detect default desktop mode
+ */
+qboolean
+GL3_GetDesktopMode(int *pwidth, int *pheight)
+{
+	// Declare display mode structure to be filled in.
+	SDL_DisplayMode current;
+
+	// We can't get desktop where we start, so use first desktop
+	if(SDL_GetDesktopDisplayMode(0, &current) != 0)
```
Comment (DanielGibson) [inline src/client/refresh/gl3/gl3_sdl.c]: can't you use https://wiki.libsdl.org/SDL_GetWindowDisplayIndex to get the desktop the window is currently on?

Comment (0lvin) [reply src/client/refresh/gl3/gl3_sdl.c]: Window is already destroyed when code run GetDesktopMode, I have moved detection code to client/vid/glimp_sdl.c as more reliable place for get window (f0dc7bd).

File: src/client/menu/menu.c [hunk 3]
```diff
@@ -1559,6 +1559,11 @@ static const char *idcredits[] = {
 	"Marty Stratton",
 	"Henk Hartong",
 	"",
+	"+Patches authors:"
+	"PGM",
+	"PMM",
```
Comment (Yamagi) [inline src/client/menu/menu.c]: Do we really need to add PGM and PMM here? That are Peter Mack and Patrick Magruder, already credited in the Ground Zero sections. That just did some changes to the client and the renderes to add support for various new effects used in Ground Zero, mostly the weapons and the new power ups.

File: src/client/refresh/gl3/gl3_sdl.c (hunk 2 above)
Comment (0lvin) [reply src/client/refresh/gl3/gl3_sdl.c]: @DanielGibson I have also added code for save window position, in other case window will always recreated/moved to first desktop.

File: src/client/menu/menu.c (hunk 3 above)
Comment (0lvin) [reply src/client/menu/menu.c]: @Yamagi I tried to save authors of patches, and if they are in list already, i have removed it from changes.

File: src/client/refresh/soft/sw_main.c [hunk 4]
```diff
@@ -1469,6 +1470,10 @@ static int RE_PrepareForWindow(void)
 	return flags;
 }
 
+// struct for save refexport callbacks, copy of re struct from main file
+// used different variable name for prevent confusion
+static refexport_t	refexport;
```
Comment (DanielGibson) [inline src/client/refresh/soft/sw_main.c]: What's the point of this change?
refexport is only used in GetRefAPI() and returned by value, I don't understand why it should be a global/static variable

Comment (0lvin) [reply src/client/refresh/soft/sw_main.c]: @DanielGibson Oops, thank you, i will fix.

**Comment by Yamagi** [comment]
And now I'm back to the Quake II business. Merged. Thank you and keep up the good work. :)

**Pull request merged**